### ISSUE TITLE
Docker-desktop for mac stuck in "Kubernetes is starting.." state

### ISSUE THREAD

**Issue by jskelcy** [docker/for-mac#3594]
- [x ] I have tried with the latest version of my channel (Stable or Edge)
  - [ x] I have uploaded Diagnostics
  - Diagnostics ID: 10C21740-9C12-4B90-A659-E1ED6F894FF5/20190325220912

### Expected behavior
Kubernetes starts
### Actual behavior
Kubernetes is stuck in starting
### Information

  - macOS Version: 10.13.6

### Diagnostic logs

```
Docker for Mac: 2.0.3.0

time="2019-03-25T18:09:12-04:00" level=info msg="Start gathering list of 72 elements (timeout=10m0s)" type=list
time="2019-03-25T18:09:12-04:00" level=info msg="Executing: [sw_vers -productVersion]" type=cmd
time="2019-03-25T18:09:12-04:00" level=info msg="Executing: [sw_vers -buildVersion]" type=cmd
time="2019-03-25T18:09:12-04:00" level=info msg="Executing: [uname -r]" type=cmd
time="2019-03-25T18:09:13-04:00" level=info msg="Executing: [sysctl kern.boottime]" type=cmd
time="2019-03-25T18:09:13-04:00" level=info msg="Executing: [sysctl hw.memsize]" type=cmd
time="2019-03-25T18:09:13-04:00" level=info msg="Executing: [vm_stat]" type=cmd
time="2019-03-25T18:09:13-04:00" level=info msg="Executing: [sysctl machdep.cpu]" type=cmd
time="2019-03-25T18:09:13-04:00" level=info msg="Executing: [uname -a]" type=cmd
time="2019-03-25T18:09:13-04:00" level=info msg="Executing: [ps uax]" type=cmd
time="2019-03-25T18:09:13-04:00" level=info msg="Executing: [dmesg]" type=cmd
time="2019-03-25T18:09:13-04:00" level=warning msg="[dmesg] exit status 1" type=cmd
time="2019-03-25T18:09:13-04:00" level=warning msg="[dmesg]: Unable to obtain kernel buffer: Operation not permitted\nusage: sudo dmesg\n" type=cmd
time="2019-03-25T18:09:13-04:00" level=info msg="Executing: [id]" type=cmd
time="2019-03-25T18:09:13-04:00" level=info msg="Executing: [sysctl -a]" type=cmd
time="2019-03-25T18:09:13-04:00" level=info msg="Executing: [mount]" type=cmd
time="2019-03-25T18:09:13-04:00" level=info msg="Executing: [df -h]" type=cmd
time="2019-03-25T18:09:13-04:00" level=info msg="Executing: [launchctl print-disabled system]" type=cmd
time="2019-03-25T18:09:13-04:00" level=info msg="Executing: [system_profiler SPHardwareDataType]" type=cmd
time="2019-03-25T18:09:13-04:00" level=info msg="Executing: [defaults read /Library/Preferences/com.apple.alf.plist]" type=cmd
time="2019-03-25T18:09:13-04:00" level=info msg="Executing: [ifconfig -a]" type=cmd
time="2019-03-25T18:09:13-04:00" level=info msg="Executing: [netstat -ln]" type=cmd
time="2019-03-25T18:09:13-04:00" level=info msg="Executing: [netstat -mm]" type=cmd
time="2019-03-25T18:09:13-04:00" level=info msg="Executing: [netstat -nr]" type=cmd
time="2019-03-25T18:09:13-04:00" level=info msg="Gathering: /etc/resolv.conf" type=file
time="2019-03-25T18:09:13-04:00" level=info msg="Gathering: /etc/hosts" type=file
time="2019-03-25T18:09:13-04:00" level=info msg="Executing: [scutil --dns]" type=cmd
time="2019-03-25T18:09:13-04:00" level=info msg="Executing: [scutil --nc list]" type=cmd
time="2019-03-25T18:09:13-04:00" level=info msg="Executing: [scutil --proxy list]" type=cmd
time="2019-03-25T18:09:13-04:00" level=info msg="Executing: [system_profiler SPFirewallDataType]" type=cmd
time="2019-03-25T18:09:13-04:00" level=info msg="HTTP GET: localhost /v1.24/version" type=http
time="2019-03-25T18:09:13-04:00" level=info msg="HTTP GET: localhost /v1.24/info" type=http
time="2019-03-25T18:09:13-04:00" level=info msg="HTTP GET: localhost /v1.24/containers/json?all=1" type=http
time="2019-03-25T18:09:13-04:00" level=info msg="HTTP GET: localhost /v1.24/nodes" type=http
time="2019-03-25T18:09:13-04:00" level=info msg="HTTP GET: localhost /v1.24/services" type=http
time="2019-03-25T18:09:13-04:00" level=info msg="HTTP GET: localhost /v1.24/networks" type=http
time="2019-03-25T18:09:13-04:00" level=info msg="Executing: [/bin/sh -c docker network inspect $(docker network ls -q)]" type=cmd
time="2019-03-25T18:09:14-04:00" level=info msg="Executing: [/usr/local/bin/kubectl.docker --context docker-for-desktop --namespace docker describe pods]" type=cmd
time="2019-03-25T18:09:14-04:00" level=info msg="Executing: [/usr/local/bin/kubectl.docker --context docker-for-desktop --namespace docker describe daemonset]" type=cmd
time="2019-03-25T18:09:14-04:00" level=info msg="Executing: [/usr/local/bin/kubectl.docker --context docker-for-desktop --namespace docker describe replicaset]" type=cmd
time="2019-03-25T18:09:14-04:00" level=info msg="Executing: [/usr/local/bin/kubectl.docker --context docker-for-desktop --namespace docker describe deployments]" type=cmd
time="2019-03-25T18:09:14-04:00" level=info msg="Executing: [/usr/local/bin/kubectl.docker --context docker-for-desktop --namespace kube-system describe pods]" type=cmd
time="2019-03-25T18:09:14-04:00" level=info msg="Executing: [/usr/local/bin/kubectl.docker --context docker-for-desktop --namespace kube-system describe daemonset]" type=cmd
time="2019-03-25T18:09:14-04:00" level=info msg="Executing: [/usr/local/bin/kubectl.docker --context docker-for-desktop --namespace kube-system describe replicaset]" type=cmd
time="2019-03-25T18:09:14-04:00" level=info msg="Executing: [/usr/local/bin/kubectl.docker --context docker-for-desktop --namespace kube-system describe deployments]" type=cmd
time="2019-03-25T18:09:15-04:00" level=info msg="Executing: [/bin/sh -c /usr/bin/log show --debug --info --style syslog --last 1d --predicate 'process matches \".*[Dd]ocker.*\" || (process in {\"kernel\", \"launchservicesd\", \"taskgated-helper\"} && eventMessage contains[c] \"docker\")']" type=cmd
time="2019-03-25T18:09:18-04:00" level=info msg="Executing: [syslog -F $((Time)(JZ)) $Host $(Sender)[$(Facility)][$(PID)] <$((Level)(str))>: $Message -o -k Sender Seq docker -o -k Sender Seq Docker -o -k Message Seq docker -o -k Message Seq Docker]" type=cmd
time="2019-03-25T18:09:18-04:00" level=warning msg="[syslog -F $((Time)(JZ)) $Host $(Sender)[$(Facility)][$(PID)] <$((Level)(str))>: $Message -o -k Sender Seq docker -o -k Sender Seq Docker -o -k Message Seq docker -o -k Message Seq Docker]: NOTE:  Most system logs have moved to a new logging system.  See log(1) for more information.\n" type=cmd
time="2019-03-25T18:09:18-04:00" level=info msg="Executing: [/bin/sh -c lsof -n -P -p $(pgrep -o com.docker.driver.amd64-linux)]" type=cmd
time="2019-03-25T18:09:19-04:00" level=info msg="Executing: [/bin/sh -c lsof -n -P -p $(pgrep -o com.docker.hyperkit)]" type=cmd
time="2019-03-25T18:09:19-04:00" level=info msg="Executing: [/bin/sh -c lsof -n -P -p $(pgrep -o com.docker.osxfs)]" type=cmd
time="2019-03-25T18:09:19-04:00" level=info msg="Executing: [/bin/sh -c lsof -n -P -p $(pgrep -o com.docker.supervisor)]" type=cmd
time="2019-03-25T18:09:19-04:00" level=info msg="Executing: [/bin/sh -c lsof -n -P -p $(pgrep -o com.docker.vpnkit)]" type=cmd
time="2019-03-25T18:09:19-04:00" level=info msg="Executing: [/bin/sh -c lsof -n -P -p $(pgrep -o com.docker.vmnetd)]" type=cmd
time="2019-03-25T18:09:19-04:00" level=warning msg="[/bin/sh -c lsof -n -P -p $(pgrep -o com.docker.vmnetd)] exit status 1" type=cmd
time="2019-03-25T18:09:19-04:00" level=info msg="Executing: [/usr/local/bin/kubectl.docker --context docker-desktop version ]" type=cmd
time="2019-03-25T18:09:19-04:00" level=info msg="Executing: [/usr/local/bin/kubectl.docker --context docker-desktop config get-contexts]" type=cmd
time="2019-03-25T18:09:19-04:00" level=info msg="Executing: [/usr/local/bin/kubectl.docker --context docker-desktop config current-context]" type=cmd
time="2019-03-25T18:09:19-04:00" level=info msg="Executing: [/usr/local/bin/kubectl.docker --context docker-desktop get nodes]" type=cmd
time="2019-03-25T18:09:19-04:00" level=warning msg="[/usr/local/bin/kubectl.docker --context docker-desktop get nodes]: No resources found.\n" type=cmd
time="2019-03-25T18:09:19-04:00" level=info msg="Executing: [/usr/local/bin/kubectl.docker --context docker-desktop get namespaces]" type=cmd
time="2019-03-25T18:09:20-04:00" level=info msg="Executing: [/usr/local/bin/kubectl.docker --context docker-desktop describe nodes]" type=cmd
time="2019-03-25T18:09:20-04:00" level=info msg="Executing: [/usr/local/bin/kubectl.docker --context docker-desktop describe pods --all-namespaces]" type=cmd
time="2019-03-25T18:09:20-04:00" level=info msg="Executing: [/usr/local/bin/kubectl.docker --context docker-desktop cluster-info dump]" type=cmd
time="2019-03-25T18:09:20-04:00" level=info msg="Executing: [/usr/local/bin/kubectl.docker --context docker-desktop describe services --all-namespaces]" type=cmd
time="2019-03-25T18:09:20-04:00" level=info msg="Executing: [/usr/local/bin/kubectl.docker --context docker-desktop describe storageclass,pv,pvc]" type=cmd
time="2019-03-25T18:09:20-04:00" level=info msg="Executing: [ls -l /Users/jskelc/Library/Containers/com.docker.docker/Data/vms/0/Docker.raw]" type=cmd
time="2019-03-25T18:09:20-04:00" level=info msg="Gathering: /Users/jskelc/Library/Group Containers/group.com.docker/DockerAppStderr.txt" type=file
time="2019-03-25T18:09:20-04:00" level=info msg="Gathering: /Users/jskelc/Library/Group Containers/group.com.docker/already-enabled-features.json" type=file
time="2019-03-25T18:09:20-04:00" level=info msg="Gathering: /Users/jskelc/Library/Group Containers/group.com.docker/dhcp.json" type=file
time="2019-03-25T18:09:20-04:00" level=info msg="Gathering: /Users/jskelc/Library/Group Containers/group.com.docker/features.json" type=file
time="2019-03-25T18:09:20-04:00" level=info msg="Gathering: /Users/jskelc/Library/Group Containers/group.com.docker/http_proxy.json" type=file
time="2019-03-25T18:09:20-04:00" level=info msg="Gathering: /Users/jskelc/Library/Group Containers/group.com.docker/settings.json" type=file
time="2019-03-25T18:09:20-04:00" level=warning msg="/Users/jskelc/Library/Preferences/group.com.docker.pliststat /Users/jskelc/Library/Preferences/group.com.docker.plist: no such file or directory" type=file
time="2019-03-25T18:09:20-04:00" level=warning msg="exit status 1" type=directorylist
time="2019-03-25T18:09:20-04:00" level=info msg="Gathering: /Users/jskelc/.docker/config.json" type=redacted-json
time="2019-03-25T18:09:20-04:00" level=info msg="Gathering: /Users/jskelc/.docker/daemon.json" type=redacted-json
time="2019-03-25T18:09:20-04:00" level=warning msg="/Library/Application Support/Docker/DockerDesktop/version-packs/version-packs.jsonstat /Library/Application Support/Docker/DockerDesktop/version-packs/version-packs.json: no such file or directory" type=file
time="2019-03-25T18:09:20-04:00" level=info msg="Getting Linux Diagnostics via a virtio socket: hyperkit:vms/0:62374" type=virtsock
time="2019-03-25T18:09:55-04:00" level=info msg="All 72 gatherers finished" type=list
```

### Steps to reproduce the behavior

  1. Enable Kuberenetes

**Comment by jskelcy** [comment]
There are a number of other troublesome logs in the `log.txt` which was generated by the diagnostics run including: 
```
2019-03-25 17:04:03.747449-0400  localhost com.docker.driver.amd64-linux[70172]: cannot read sysevent: EOF
```
and 
```
2019-03-25 17:14:29.384308-0400  localhost com.docker.driver.amd64-linux[70172]: Cannot list nodes: Get https://localhost:6443/api/v1/nodes: EOF
```

**Comment by ulrich** [comment]
Hello @jskelcy 

If it is possible can you reset to default settings and try again?

Reset will destroy your cluster and images, so don't forget to save settings and images.

**Comment by jskelcy** [comment]
I should have updated this ticket. That is exactly what I did and it worked.

**Comment by ulrich** [comment]
Thank you @jskelcy

**Comment by kkzz8888** [comment]
Is there a way to recover the k8s cluster without having to reset k8s cluster (which will cause loss of all current k8s resource/data)? that is a lot of work that I have done including a database running posgresql within k8s.

**Comment by fksimon** [comment]
I experienced the same problem.   Only option I had was a full reset of docker desktop which was quite annoying.   And because it was stuck in starting I couldn't just disable the kubernetes cluster and reenable, that option was grayed out.

**Comment by fescobar** [comment]
https://github.com/docker/for-mac/issues/3649#issuecomment-497441158

**Comment by StephanAndre** [comment]
I was able to disable Kubernetes as follows without full reset (Status: Docker running / Kubernetes starting, Preferences all grayed out):

From the docker icon, select Restart.
Quickly open the preferences. Most likely it is not grayed out now.
Quickly select Kubernetes, turn off "Enable Kubernetes", press Apply & Restart.
This needs to be done within a few seconds.

When back, only Docker is running. Preferences can be used again.

**Comment by n1t1nv3rma** [comment]
StephanAndre doesn't work anymore!

**Comment by croydondias** [comment]
As per https://github.com/docker/for-mac/issues/3649#issuecomment-540953659 , dlete the pki folder fixed this issue on Docker desktop version **2.2.0.5**


`rm -rf ~/Library/Group\ Containers/group.com.docker/pki/`

**Comment by Cephei** [comment]
Thank you @croydondias. I did a Reset to Defaults, but that didn't solve the issue. Removing /pki did.
